lib/core/test/unit/Cardano/Wallet/DB/Arbitrary.hs

```diff
@@ -471,6 +484,35 @@ defaultSeqStatePrefix = DerivationPrefix
     , minBound
     )
 
+instance Arbitrary ScriptHash where
+    arbitrary =
+        pure $ ScriptHash (BS.replicate 28 0)
+
+genVerificationKeyPool
+    :: ShelleyKey 'AccountK XPub
+    -> Gen (VerificationKeyPool ShelleyKey)
+genVerificationKeyPool accXPub = do
+    nVerKeys <- choose (5,10)
+    let minIndex = getIndex @'Soft minBound
+    let toVerKeyHash ix =
+            hashVerificationKey $
+            deriveVerificationKey accXPub
+            (toEnum (fromInteger $ toInteger $ minIndex + ix))
+    verKeysIxs <- L.nub <$> vectorOf nVerKeys (choose (0, 15))
+    let nVerKeys' = L.length verKeysIxs
+    let setUsed ix =
+            if ix `elem` verKeysIxs then
+                Used
+            else
+                Unused
+    let indexedKeysMap = map (\ix -> (toVerKeyHash ix, (Index ix, setUsed ix)))
+            [0 .. maximum verKeysIxs]
+    knownScripts <- vectorOf nVerKeys' arbitrary
+    let knownScriptsMap =
+            zipWith (\s k -> (s,[k])) knownScripts (Index <$> verKeysIxs)
+    pure $ unsafeVerificationKeyPool accXPub defaultAddressPoolGap
+        (Map.fromList indexedKeysMap) (Map.fromList knownScriptsMap)
```

Comment:
This generator is a bit hard to follow. In particular, what properties does it have and what sort of pool does it generate? Since it is merely for the database properties, do we really need something that complicated? 

For example, the database generator for `AddressPool` is simply:

```hs
instance Typeable chain => Arbitrary (AddressPool chain ShelleyKey) where
    arbitrary = pure $ mkAddressPool @'Mainnet arbitrarySeqAccount minBound mempty
```

which really is sufficient for testing serialization and deserialization + various database interactions.

Reply:
I wanted it to be close to reality, to have our verification key for discovered script. or even more than one. And check storing such a state and retrieving it and make sure I deal with array_ix properly. There was a time I got it wrong